lib/modules/cpu.go

```diff
@@ -0,0 +1,82 @@
+package modules
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"github.com/davidscholberg/go-i3barjson"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// Cpu represents the configuration for the cpu block.
+type Cpu struct {
+	BlockConfigBase `yaml:",inline"`
+	Cpu             string  `yaml:"cpu"` // name of the cpu to display (from /proc/stat), empty = all
+	CritUsage       float64 `yaml:"crit_usage"`
+
+	lastIdle  uint64
+	lastTotal uint64
```

Comment:
I understand.

I think it would be great if the whole goblocks design was based around blocks that are once configured from YAML and live until the program exits. The block would then encapsulate its configuration, state and provide an interface to render itself on screen. I actually though this is the way how it worked when I coded the CPU block. I think it is more documentation/mental and probably pointer receiver problem than anything else.

I was also thinking about moving the state of the block to a global variable and the best design I can come with is this:

```
type cpuState struct {
	lastIdle  uint64
	lastTotal uint64
}

// name of the CPU -> block state
var cpuStates map[string]cpuState
```

It has the shorcoming that it does not support creating a block for the same cpu twice. It is rather theoretical problem but it points to a more higher level problem.

BTW is guaranteed that the `UpdateBlock` functions are called from one goroutine? Otherwise the state access would have to be synchronized.